ipld/src/util.rs

```diff
@@ -3,36 +3,22 @@
 
 use std::future::Future;
 
-use ahash::HashSet;
 use cid::Cid;
 use fvm_ipld_encoding::from_slice;
 
-use crate::Ipld;
-
-/// Basic trait to abstract way the hashing details to the trait implementation.
-pub trait InsertHash {
-    /// Hashes the input and inserts its hash into the underlying collection.
-    fn hash_and_insert(&mut self, data: &[u8]) -> bool;
-}
-
-impl InsertHash for HashSet<blake3::Hash> {
-    fn hash_and_insert(&mut self, data: &[u8]) -> bool {
```

Comment:
The hasher needs to be passed to this function when it's non-deterministic, it would be easier to use a struct instead